CyborgianStates/Commands/CouldEndorseCommand.cs

```diff
@@ -0,0 +1,179 @@
+
+            List<DumpNation> couldEndorse = _dumpDataService.GetWANationsByRegionName(dumpNation.RegionName); // Get all WA nations in the same region.
+            couldEndorse.RemoveAll(n => dumpNation.Endorsements.Contains(n.Name)); // Remove all nations that already have endorsements by the dumpNation.
+            couldEndorse.Remove(dumpNation); // Remove the nation itself.
+            // Convert the list of nations to a string.
+            List<string> couldEndorseNames = couldEndorse.Select(n => n.Name).ToList();
+            await SplitResponseAsync(message, dumpNation, couldEndorseNames).ConfigureAwait(true);
+
+            DumpRetrievalBackgroundService dumpService = new DumpRetrievalBackgroundService();
+            TimeSpan? hoursSinceUpdate = GetUpdateTime(dumpService, UpdateTime.Last);
+            TimeSpan? hoursUntilUpdate = GetUpdateTime(dumpService, UpdateTime.Next);
+
+            
+            _responseBuilder.Clear();
+            _responseBuilder.Success()
+                .WithTitle("Finished") // TODO: Bug likely in BaseResponseBuilder.Clear(). Title and Description are not cleared.
+                .WithDescription("") // TODO: Bug likely in BaseResponseBuilder.Clear(). Title and Description are not cleared.
+                .WithField("Datascource", "Dump", true)
+                .WithField("As of", $"{(hoursSinceUpdate.HasValue ? hoursSinceUpdate.Value.ToString("g") : "Unknown")} ago", true)
+                .WithField("Next update in", $"{(hoursUntilUpdate.HasValue ? hoursUntilUpdate.Value.ToString("g") : "Unknown")}", true)
+                .WithFooter(_config.Footer);
+        }
+        
+        
+        private async Task SplitResponseAsync(Message message, DumpNation nation, List<string> endorsable)
+        {
+            const int limit = 3800; // Less than 4096 description limit to accommodate for ", " in string.Join.
+            
+            Stack<string> endorse = new Stack<string>(endorsable.Select(n => $"[{n}](https://www.nationstates.net/nation={n})"));
+            
+            int totalChars = endorse.Sum(n => n.Length);
+            if (totalChars < limit) // Determine if we need to split the response at all.
+            {
+                _responseBuilder.Success()
+                    .WithTitle($"{nation.Name} could endorse {endorsable.Count} more nations.")
+                    .WithDescription(string.Join(", ", endorse))
+                    .WithFooter(_config.Footer);
+                return;
+            }
+            
+            // Send out embeds in chunks of 3800 characters from the stack.
+            while (endorse.Count > 0)
+            {
+                // Get the next chunk of 3800 characters.
+                List<string> chunk = new List<string>();
+                int chunkSize = 0;
+                while (chunkSize < limit && endorse.Count > 0)
+                {
+                    chunk.Add(endorse.Pop());
+                    chunkSize += chunk.Last().Length;
```

Comment:
If you would store the result of `endorse.Pop()` you wouldn't need to call Last() on the chunk to get it's length.

Reply:
Is this for performance reasons? I've modified it now for the next push.